backend/index.js

```diff
@@ -86,11 +97,7 @@ app.post("/logout", (req, res) => {
 
 // Create Post Route
 app.post("/createPost", uploadMiddleware.single('file'), async (req, res, next) => {
-  const { originalname, path } = req.file;
-  const parts = originalname.split(".");
-  const ext = parts[parts.length - 1];
-  const newPath = `${parts[0]}.${ext}`;
-  fs.renameSync(path, newPath);
+  const { filename } = req.file;
```

Comment:
Here `filename` is the name we generated in the **multer** config above.
This line specifically
```js
cb(null, `${Date.now()}.${ext}`);
```